projects/kit/components/input-date/input-date.directive.ts

```diff
@@ -118,6 +118,15 @@ export abstract class TuiInputDateBase<
         this.max.set(max instanceof TuiDay ? max : this.options.max);
     }
 
+    public override writeValue(value: T | null): void {
+        const reset = this.control.pristine && this.control.untouched && !value;
+
+        if (value !== this.value() || reset) {
+            super.writeValue(value);
+            this.textfield.value.set(this.stringify(this.value()));
```

Comment:
Failed case with `writeValue` !

https://taiga-ui.dev/components/input-date#datalist

<img width="2016" height="176" alt="image" src="https://github.com/user-attachments/assets/18f946e5-0401-4a2f-b3c5-477fbe0a6217" />

Reply:
Rebase this branch after merging the following PR:
* https://github.com/taiga-family/taiga-ui/pull/11460